src/hn/core.clj

```diff
@@ -15,9 +16,11 @@
       (fn [response]
         (dispatch! (assoc (:on-response v) :response response)))
       (fn [exception]
-        (dispatch! (assoc (:on-exception v) :exception exception))))
+        (dispatch! (assoc (:on-exception v) ::event/type ::event/exception
+                                            :exception exception))))
     (catch Exception e
-      (dispatch! (assoc (:on-exception v) :exception e)))))
+      (dispatch! (assoc (:on-exception v) ::event/type ::event/exception
```

Comment:
This does not look right to me: HTTP effect should be only about executing HTTP requests and returning back the control to the callee. There is a problem with the code, but it's not here: the actual usage of HTTP effect has to specify `:on-exception` event explicitly.

Reply:
Ah I see - I think I've addressed this, let me know what you think.